src/link/Elf.zig

```diff
@@ -1472,6 +1473,10 @@ fn linkWithLLD(self: *Elf, comp: *Compilation, prog_node: *std.Progress.Node) !v
             try argv.append("--emit-relocs");
         }
 
+        if (self.base.options.relocatable) {
+            try argv.append("--relocatable");
```

Comment:
Note line 1419. Unfortunately for this particular change it won't be this easy; it requires changes to the CLI, Compilation.zig, and may require reworking some additional logic. This is related to #7547 and #7094. Note also that COFF and MachO linkers do not support this flag.

Reply:
Uh oh. Thanks.

I will file an issue instead. As far as my (very limited) scope is concerned, if `-r` is passed to `zig cc`, it should be passed to the linker.